pkg/service/presentation/service.go

```diff
@@ -0,0 +1,109 @@
+
+	"github.com/tbd54566975/ssi-service/config"
+	"github.com/tbd54566975/ssi-service/internal/util"
+	"github.com/tbd54566975/ssi-service/pkg/service/framework"
+	presentationstorage "github.com/tbd54566975/ssi-service/pkg/service/presentation/storage"
+	"github.com/tbd54566975/ssi-service/pkg/storage"
+)
+
+type Service struct {
+	storage presentationstorage.Storage
+	config  config.PresentationServiceConfig
+}
+
+func (s Service) Type() framework.Type {
+	return framework.Presentation
+}
+
+func (s Service) Status() framework.Status {
+	ae := sdkutil.NewAppendError()
+	if s.storage == nil {
+		ae.AppendString("no storage configured")
+	}
+	if !ae.IsEmpty() {
+		return framework.Status{
+			Status:  framework.StatusNotReady,
+			Message: fmt.Sprintf("presentation service is not ready: %s", ae.Error().Error()),
+		}
+	}
+	return framework.Status{Status: framework.StatusReady}
+}
+
+func (s Service) Config() config.PresentationServiceConfig {
+	return s.config
+}
+
+func NewPresentationDefinitionService(config config.PresentationServiceConfig, s storage.ServiceStorage) (*Service, error) {
+	presentationStorage, err := presentationstorage.NewPresentationStorage(s)
+	if err != nil {
+		errMsg := "could not instantiate storage for the presentation definition service"
+		return nil, util.LoggingErrorMsg(err, errMsg)
+	}
+	service := Service{
+		storage: presentationStorage,
+		config:  config,
+	}
+	if !service.Status().IsReady() {
+		return nil, errors.New(service.Status().Message)
+	}
+	return &service, nil
+}
+
+// CreatePresentationDefinition houses the main service logic for presentation definition creation. It validates the input, and
+// produces a presentation definition value that conforms with the PresentationDefinition specification.
+func (s Service) CreatePresentationDefinition(request CreatePresentationDefinitionRequest) (*CreatePresentationDefinitionResponse, error) {
+	logrus.Debugf("creating presentation definition: %+v", request)
+
+	if !request.IsValid() {
+		errMsg := fmt.Sprintf("invalid create presentation definition request: %+v", request)
+		return nil, util.LoggingNewError(errMsg)
+	}
+
+	if err := exchange.IsValidPresentationDefinition(request.PresentationDefinition); err != nil {
+		return nil, util.LoggingErrorMsg(err, "provided value is not a valid presentation definition")
+	}
+
+	storedPresentation := presentationstorage.StoredPresentation{ID: request.PresentationDefinition.ID, PresentationDefinition: request.PresentationDefinition}
+
+	if err := s.storage.StorePresentation(storedPresentation); err != nil {
+		return nil, util.LoggingErrorMsg(err, "could not store presentation")
+	}
+
+	return &CreatePresentationDefinitionResponse{
+		PresentationDefinition: storedPresentation.PresentationDefinition,
+	}, nil
+}
+
+func (s Service) GetPresentationDefinition(request GetPresentationDefinitionRequest) (*GetPresentationDefinitionResponse, error) {
+	logrus.Debugf("getting presentation definition: %s", request.ID)
```

Comment:
are these debug logs on purpose?

Reply:
think they're fine we may want to add more